### ISSUE TITLE
Invalid URLs shouldn't cause errors

### ISSUE THREAD

**Issue by captn3m0** [r7kamura/weneedfeed#18]
If the `item_link_selector` refers to a relative URL, the build fails.

```
/usr/lib/ruby/3.0.0/uri/rfc3986_parser.rb:67:in `split': bad URI(is not URI?): "/PDF/cts/CTS-Northern -GRID-Uniform-holiday-list-2016.pdf" (URI::InvalidURIError)
	from /usr/lib/ruby/3.0.0/uri/rfc3986_parser.rb:72:in `parse'
	from /usr/lib/ruby/3.0.0/uri/rfc3986_parser.rb:106:in `convert_to_uri'
	from /usr/lib/ruby/3.0.0/uri/generic.rb:1100:in `merge'
	from /usr/lib/ruby/3.0.0/uri/rfc3986_parser.rb:78:in `inject'
	from /usr/lib/ruby/3.0.0/uri/rfc3986_parser.rb:78:in `join'
	from /usr/lib/ruby/3.0.0/uri/common.rb:208:in `join'
	from ruby/3.0.0/gems/weneedfeed-0.9.2/lib/weneedfeed/item.rb:86:in `link'
	from ruby/3.0.0/gems/weneedfeed-0.9.2/templates/show_feed.xml.erb:14:in `block in __tilt_1520'
	from ruby/3.0.0/gems/weneedfeed-0.9.2/templates/show_feed.xml.erb:11:in `each'
	from ruby/3.0.0/gems/weneedfeed-0.9.2/templates/show_feed.xml.erb:11:in `__tilt_1520'
	from ruby/3.0.0/gems/tilt-2.0.10/lib/tilt/template.rb:170:in `call'
	from ruby/3.0.0/gems/tilt-2.0.10/lib/tilt/template.rb:170:in `evaluate'
	from ruby/3.0.0/gems/tilt-2.0.10/lib/tilt/template.rb:109:in `render'
	from ruby/3.0.0/gems/hibana-0.2.0/lib/hibana/view.rb:37:in `render_partial'
	from ruby/3.0.0/gems/hibana-0.2.0/lib/hibana/view.rb:21:in `to_s'
	from ruby/3.0.0/gems/weneedfeed-0.9.2/lib/weneedfeed/controllers/show_feed.rb:35:in `call'
	from ruby/3.0.0/gems/hibana-0.2.0/lib/hibana/controller.rb:13:in `call'
	from ruby/3.0.0/gems/hanami-router-2.0.0.alpha6/lib/hanami/router.rb:105:in `call'
	from ruby/3.0.0/gems/hibana-0.2.0/lib/hibana/application.rb:42:in `call'
	from ruby/3.0.0/gems/weneedfeed-0.9.2/lib/weneedfeed/application.rb:21:in `call'
	from ruby/3.0.0/gems/rack-capture-0.4.0/lib/rack/capture.rb:56:in `get_response'
	from ruby/3.0.0/gems/rack-capture-0.4.0/lib/rack/capture.rb:32:in `call'
	from ruby/3.0.0/gems/rack-capture-0.4.0/lib/rack/capture.rb:11:in `call'
	from ruby/3.0.0/gems/weneedfeed-0.9.2/lib/weneedfeed/capture.rb:34:in `block in call'
	from ruby/3.0.0/gems/weneedfeed-0.9.2/lib/weneedfeed/capture.rb:33:in `each'
	from ruby/3.0.0/gems/weneedfeed-0.9.2/lib/weneedfeed/capture.rb:33:in `call'
	from ruby/3.0.0/gems/weneedfeed-0.9.2/lib/weneedfeed/capture.rb:18:in `call'
	from ruby/3.0.0/gems/weneedfeed-0.9.2/lib/weneedfeed/command.rb:39:in `build'
	from ruby/3.0.0/gems/thor-1.2.1/lib/thor/command.rb:27:in `run'
	from ruby/3.0.0/gems/thor-1.2.1/lib/thor/invocation.rb:127:in `invoke_command'
	from ruby/3.0.0/gems/thor-1.2.1/lib/thor.rb:392:in `dispatch'
	from ruby/3.0.0/gems/thor-1.2.1/lib/thor/base.rb:485:in `start'
	from ruby/3.0.0/gems/weneedfeed-0.9.2/exe/weneedfeed:6:in `<top (required)>'
	from ruby/3.0.0/bin/weneedfeed:25:in `load'
	from ruby/3.0.0/bin/weneedfeed:25:in `<top (required)>'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/lib/bundler/cli/exec.rb:58:in `load'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/lib/bundler/cli/exec.rb:58:in `kernel_load'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/lib/bundler/cli/exec.rb:23:in `run'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/lib/bundler/cli.rb:483:in `exec'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/lib/bundler/cli.rb:31:in `dispatch'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/lib/bundler/cli.rb:25:in `start'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/exe/bundle:48:in `block in <top (required)>'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	from /usr/lib/ruby/gems/3.0.0/gems/bundler-2.3.10/exe/bundle:36:in `<top (required)>'
	from /usr/sbin/bundle:25:in `load'
	from /usr/sbin/bundle:25:in `<main>'
```

**Comment by r7kamura** [comment]
@captn3m0 I think there is something else wrong with it because this tool supports relative URL.

https://github.com/r7kamura/weneedfeed/blob/f57480885ba4a019484e2c17783408900dd42a27/spec/weneedfeed/item_spec.rb#L22-L29

The error message appears to be saying that the URL path has invalid format:

> /usr/lib/ruby/3.0.0/uri/rfc3986_parser.rb:67:in `split': bad URI(is not URI?): "/PDF/cts/CTS-Northern -GRID-Uniform-holiday-list-2016.pdf" (URI::InvalidURIError)

I think it is invalid to include spaces:

```
irb(main):001:0> URI.join("http://example.com", "/a b")
Traceback (most recent call last):
        26: from /home/r7kamura/.rbenv/versions/2.7.2/bin/irb:23:in `<main>'                                                                                
        25: from /home/r7kamura/.rbenv/versions/2.7.2/bin/irb:23:in `load'                                                                                  
        24: from /home/r7kamura/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'                                   
         7: from (irb):5:in `<main>'                                                                                                                        
         6: from /home/r7kamura/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/common.rb:271:in `join'                                                            
         5: from /home/r7kamura/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/rfc3986_parser.rb:89:in `join'                                                     
         4: from /home/r7kamura/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/rfc3986_parser.rb:89:in `inject'                                                   
         3: from /home/r7kamura/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/generic.rb:1101:in `merge'                                                         
         2: from /home/r7kamura/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/rfc3986_parser.rb:117:in `convert_to_uri'                                          
         1: from /home/r7kamura/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/rfc3986_parser.rb:73:in `parse'                                                    
/home/r7kamura/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/rfc3986_parser.rb:67:in `split': bad URI(is not URI?): "/a b" (URI::InvalidURIError)
```

Either the URL in the target HTML source is incorrect, or this library has encoded it incorrectly. Should we do something about it on this library side?

**Comment by captn3m0** [comment]
The HTML itself is invalid: https://validator.w3.org/nu/?doc=https%3A%2F%2Fwww.npci.org.in%2Fwhat-we-do%2Fcts%2Fcirculars

But such href attributes exist on the internet, weneedfeed shouldn't break on encountering these.

**Comment by r7kamura** [comment]
Also fixed at https://github.com/r7kamura/weneedfeed/pull/25 and released v0.11.0 👍
